packages/hub/node-tests/@core/compiler-test.ts

```diff
@@ -484,6 +484,31 @@ if (process.env.COMPILER) {
         `);
       });
 
+      it('can compile schema class promise functions', async function () {
+        let { compiled } = await cards.load(`${realm}person`);
+        let source = getFileCache().getModule(compiled.schemaModule.global, 'browser');
+        expect(source).to.containsSource(`
+          then(resolve) {
+            let fields = ["lastName", "aboutMe", "fullName"];
+            return Promise.all(fields.map(field => this[field])).then(values => resolve(Object.fromEntries(fields.map((field, i) => [field, values[i]]))));
+          }
+        `);
+        expect(source).to.containsSource(`
+          catch(err) {
+            if (err) {
+              err();
+            }
+          }
+        `);
```

Comment:
This doesn't seem correct...why would we call the error? Also, whats the plan for this?

Reply:
This is for the consumer of the promise to provide an error callback that is evaluated when our promise rejects. I think the way this is really implemented, according to MDN, is that it is just an alias for a `.then(undefined, reject)`. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch. I wasn't really fleshing this out. but i can do that.

Reply:
Something like this might work out nicely: 
```js
          get #allValues() {
            return Promise.all(this.#allFields.map(field => this[field]))
          }
          get #allFields() {
            return ['lastName', 'aboutMe', 'fullName'];
          }
          then(resolve, reject) {
            this.#allValues
              .then(values => {
                if (resolve) {
                  resolve(Object.fromEntries(this.#allFields.map((field, i) => [field, values[i]])))
                }
              })
              .catch(err => { 
                if( reject) { reject(err) } 
              });
          }
          catch(err) {
            this.then(undefined, err);    
          }
```

Reply:
in terms of your question "why would we call the error". It's not that "we" as in our schema class would ever call the error--but rather if we want to behave like a promise we need to provide a catch that allows consumers of the promise to handle errors. For instance, the process of getting a value from one of our fields (the `Promise.all`) may trigger an error in a computed (it is user code after all--as well as our own hub code may encounter an error. i believe for `async` functions, when you wrap a `try/catch` around an `await`, that is another way to provide a `.catch` error callback to the promise. Node should know how to go between these 2 different patterns for error handling. In terms of being good citizens, if we are going to say that our Schema classes implement a `Promise` interface, it would behoove us to provide all the promise capabilities. We will also need to do this for our `FieldGetters` as well.

Reply:
turns out private methods pull in lots of babel custom code and significantly rewrites our transpiled schema. In the interest of nice clean schema sources, I'm gonna just use normal class methods....